Mesh_3/include/CGAL/Mesh_3/tet_soup_to_c3t3.h

```diff
@@ -0,0 +1,447 @@
+    // assign cells to vertices
+    for(int j=0; j<4; ++j)
+    {
+      if(vs[j]->cell() == Cell_handle())
+        vs[j]->set_cell(c);
+    }
+
+    // build the map used for adjacency later
+    for(int j=0; j<4; ++j)
+    {
+      add_facet_to_incident_cells_map<Tr>(c, j, incident_cells_map);
+      if(border_facets.size() !=0)
+      {
+        boost::array<int,3> facet;
+        facet[0]=tet[(j+ 1) % 4];
+        facet[1]=tet[(j+ 2) % 4];
+        facet[2]=tet[(j+ 3) % 4];
+        //find the circular permutation that puts the smallest index in the first place.
+        int n0 = (std::min)((std::min)(facet[0],facet[1]), facet[2]);
+        int k=0;
+        boost::array<int,3> f;
+        do
+        {
+          f[0]=facet[(0+k)%3];
+          f[1]=facet[(1+k)%3];
+          f[2]=facet[(2+k)%3];
+          ++k;
+        }while(f[0] != n0);
+        if(border_facets.find(f) != border_facets.end())
+          c->set_surface_patch_index(j, border_facets.at(f));
+        else
+        {
+          int temp = f[2];
+          f[2]=f[1];
+          f[1]=temp;
+          if(border_facets.find(f) != border_facets.end())
+            c->set_surface_patch_index(j, border_facets.at(f));
+          else
+            c->set_surface_patch_index(j, 0);
+        }
+      }
+    }
+  }
+}
+
+template<class Tr>
+void add_infinite_facets_to_incident_cells_map(typename Tr::Cell_handle c,
+                                               int inf_vert_pos,
+                                               std::map<std::set<typename Tr::Vertex_handle>,
+                                               std::vector<std::pair<typename Tr::Cell_handle,
+                                               int> > >& incident_cells_map)
+{
+  int l = (inf_vert_pos + 1) % 4;
+  add_facet_to_incident_cells_map<Tr>(c, l, incident_cells_map);
+  l = (inf_vert_pos + 2) % 4;
+  add_facet_to_incident_cells_map<Tr>(c, l, incident_cells_map);
+  l = (inf_vert_pos + 3) % 4;
+  add_facet_to_incident_cells_map<Tr>(c, l, incident_cells_map);
+}
+
+template<class Tr>
+void build_infinite_cells(Tr& tr,
+                          const std::vector<boost::array<int,4> >& infinite_cells,
+                          const std::vector<typename Tr::Vertex_handle>& vertex_handle_vector,
+                          std::map<std::set<typename Tr::Vertex_handle>,
+                          std::vector<std::pair<typename Tr::Cell_handle,
+                          int> > >& incident_cells_map)
+{
+  typedef boost::array<int, 4>                            Tet; // 4 ids
+
+  typedef typename Tr::Vertex_handle                      Vertex_handle;
+  typedef typename Tr::Cell_handle                        Cell_handle;
+  typedef std::set<Vertex_handle>                         Facet;
+  typedef std::pair<Cell_handle, int>                     Incident_cell;
+  typedef std::map<Facet, std::vector<Incident_cell> >    Incident_cells_map;
+
+  // build the infinite cells if provided
```

Comment:
There is no reading of infinite cells anymore so the whole `for()` can be removed